llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -126,6 +126,11 @@ static cl::opt<bool> MulConstantOptimization(
              "SHIFT, LEA, etc."),
     cl::Hidden);
 
+static cl::opt<bool>
+    EnableBaseIndexUpdate("update-baseIndex", cl::init(true),
```

Comment:
Does not follow command line option naming convention

Reply:
Also this is the kind of option that will never be used by anyone. Can you just do this unconditionally

Reply:
I just added this flag for the safety incase any untest usecase occur. So that we can fall back to existing behavior. 
Sure, I will remove the flag.